internal/jimm/juju/migrationtarget.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025 Canonical.
+
+package juju
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/juju/juju/core/migration"
+	"github.com/juju/names/v5"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+// Prechecks checks that the model can be migrated to the target controller.
+// It does this by calling the method of the same name on the target Juju controller.
+// As part of all model migrations passing through JIMM, it modifies the model description
+// to replace any local user references with their external mapping.
+func (j *JujuManager) Prechecks(ctx context.Context, user *openfga.User, model migration.ModelInfo) error {
+	const op = errors.Op("jimm.Prechecks")
+
+	modelMigration := dbmodel.IncomingModelMigration{
+		ModelUUID: sql.NullString{
+			String: model.UUID,
+			Valid:  true,
+		},
+	}
+	err := j.Database.GetIncomingModelMigration(ctx, &modelMigration)
```

Comment:
Let's not call it modelMigration, I think this reads a bit clearer:
```go
func (j *jujuManager) getUserMapping(modelUUID string) (StringMap, error) {
	incomingModelMigration := dbmodel.IncomingModelMigration{
		ModelUUID: sql.NullString{
			String: model.UUID,
			Valid:  true,
		},
	}
	if err := j.Database.GetIncomingModelMigration(ctx, &incomingModelMigration); err != nil {
		return errors.E(op, fmt.Errorf("failed to get model migration %q: %w", model.UUID, err))
       }
       
       return ncomingModelMigration.UserMapping
}

````

Because logic wise we don't care about anything other than the mapping, and it'll read nicely this way.
(Inline function probably better?)